twins/nats/pubsub.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package nats
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux"
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/twins"
+	"github.com/mainflux/mainflux/twins/mqtt"
+	"github.com/mainflux/senml"
+	"github.com/nats-io/go-nats"
+)
+
+const (
+	queue = "twins"
+	input = "channel.>"
+)
+
+var mqttOp = map[string]string{
+	"stateSucc": "state/success",
+	"stateFail": "state/failure",
+}
+
+type pubsub struct {
+	natsClient *nats.Conn
+	mqttClient mqtt.Mqtt
+	logger     log.Logger
+	twins      twins.TwinRepository
```

Comment:
Because we do not want to have overhead of using a service when we only need repos. Should I use service rather? I think the latter is not a wise idea, since you update state very often. You don't want to pass by the service overhead. @nmarcetic ^

Reply:
I would also say service, why not.

Reply:
We now use the service. The business logic is also moved to service.